fix some path to sysinfo.sh on gentoo install
sysinfo.sh prepared to use /etc/conf.d/vdr.sysinfo

Joerg Bornkessel hd_brummy@gentoo.org ( 22 May 2006 )

diff -Naur sysinfo-0.1.0a.orig/script/sysinfo.sh sysinfo-0.1.0a/script/sysinfo.sh
--- sysinfo-0.1.0a.orig/script/sysinfo.sh	2006-05-22 18:53:38.770308750 +0200
+++ sysinfo-0.1.0a/script/sysinfo.sh	2006-05-22 18:54:25.325218250 +0200
@@ -1,31 +1,35 @@
 #!/bin/bash
 
+source /etc/conf.d/vdr.sysinfo
+
 case "$1" in
 	cputemp)
-		sensors | grep -i 'CPU TEMP' | cut -c 10-18 | tr -d ' '
-		exit $?
+		sensors | grep -i ${CPU_TEMP:=CPU Temp} | cut -c 10-18 | tr -d ' '
         ;;
 	cpufan)
-		sensors | grep -i 'FAN1' | cut -c 10-18 | tr -d ' '
-		exit $?
+		sensors | grep -i ${FAN_1:=FAN1} | cut -c 10-18 | tr -d ' '
         ;;
+	cputime)
+		ps -eo%C | awk '/[.]/ { a=a+$1 } ; END { print a }'
+		;;
 	mbtemp)
-		sensors | grep -i 'M/B TEMP' | cut -c 10-18 | tr -d ' '
-		exit $?
+		sensors | grep -i ${MOBO_TEMP:=M/B Temp} | cut -c 10-18 | tr -d ' '
         ;;
 	mbfan)
-		sensors | grep -i 'FAN2' | cut -c 10-18 | tr -d ' '
-		exit $?
+		sensors | grep -i ${FAN_2:=FAN2} | cut -c 10-18 | tr -d ' '
         ;;		
 	vidspace)
-		df -h | grep hda1 | cut -c 33-38 | tr -d ' '
-		exit $?
+		df -h | grep hd | grep video | awk '{ print $4}'
         ;;		
 	*)
-		echo ""
-		echo "Usage: sysinfo.sh {cputemp|cpufan|mbtemp|mbfan|vidspace}"
-		echo ""
+		echo "Usage: sysinfo.sh {cputemp|cpufan|cputime|mbtemp|mbfan|vidspace}"
 		exit 1
 		;;
 esac
+
+case ${PIPESTATUS[@]} in
+     0|'0 0'*) exit 0 ;;
+     *) exit 1 ;;
+esac
+
 exit 0
diff -Naur sysinfo-0.1.0a.orig/sysinfoosd.c sysinfo-0.1.0a/sysinfoosd.c
--- sysinfo-0.1.0a.orig/sysinfoosd.c	2006-05-22 18:53:38.770308750 +0200
+++ sysinfo-0.1.0a/sysinfoosd.c	2006-05-22 18:56:07.627611750 +0200
@@ -152,18 +152,18 @@
 	osd->DrawText(15, iTop+10, cTemp, pal[3],pal[1],font);
 	sprintf(cTemp, tr("M/B temperature:"));
 	osd->DrawText(15, iTop+40, cTemp, pal[3],pal[1],font);
-	ExecShellCmd("sysinfo.sh cputemp", cTemp);
+	ExecShellCmd("/usr/share/vdr/sysinfo/sysinfo.sh cputemp", cTemp);
 	osd->DrawText(270, iTop+10, cTemp, pal[3],pal[1],font);
-	ExecShellCmd("sysinfo.sh mbtemp", cTemp);
+	ExecShellCmd("/usr/share/vdr/sysinfo/sysinfo.sh mbtemp", cTemp);
 	osd->DrawText(270, iTop+40, cTemp, pal[3],pal[1],font);
 	// Fan rpm
 	sprintf(cTemp, tr("Fan1:"));
 	osd->DrawText(((iWidth/2)+20), iTop+10, cTemp, pal[3],pal[1],font);
 	sprintf(cTemp, tr("Fan2:"));
 	osd->DrawText(((iWidth/2)+20), iTop+40, cTemp, pal[3],pal[1],font);
-	ExecShellCmd("sysinfo.sh cpufan", cTemp);
+	ExecShellCmd("/usr/share/vdr/sysinfo/sysinfo.sh cpufan", cTemp);
 	osd->DrawText(((iWidth/2)+170), iTop+10, cTemp, pal[3],pal[1],font);
-	ExecShellCmd("sysinfo.sh mbfan", cTemp);
+	ExecShellCmd("/usr/share/vdr/sysinfo/sysinfo.sh mbfan", cTemp);
 	osd->DrawText(((iWidth/2)+170), iTop+40, cTemp, pal[3],pal[1],font);
 	// Draw line separator
 	osd->DrawRectangle(iTop+74, 0, iWidth, 1, pal[2]);
@@ -174,7 +174,7 @@
 	char cTemp[100];	// 2DO
 	sprintf(cTemp, tr("Video disk free space:"));
 	osd->DrawText(15, iTop+10, cTemp, pal[3],pal[1],font);
-	ExecShellCmd("sysinfo.sh vidspace", cTemp);
+	ExecShellCmd("/usr/share/vdr/sysinfo/sysinfo.sh vidspace", cTemp);
 	osd->DrawText(285, iTop+10, cTemp, pal[3],pal[1],font);
 	// Draw line separator
 	osd->DrawRectangle(iTop+46, 0, iWidth, 1, pal[2]);
@@ -222,7 +222,7 @@
 
 	// Calculate cpu free
 	//ExecShellCmd("CPU=0; for I in `ps -eo ""%C"" | grep ""\\.""`; do CPU=`echo $CPU $I | awk '{ print $1 + $2 }'`; done; echo $CPU", cTemp);
-	ExecShellCmd("sysinfo.sh cputime", cTemp);
+	ExecShellCmd("/usr/share/vdr/sysinfo/sysinfo.sh cputime", cTemp);
 	iPerCentCpu=atoi(cTemp);
 	if (iPerCentCpu>100) iPerCentCpu=100;
 	sprintf(cTemp, tr("CPU used:"));
